subsys/dfu/dfu_target/src/dfu_target.c

```diff
@@ -91,11 +92,13 @@ int dfu_target_init(int img_type, int img_num, size_t file_size, dfu_target_call
 	 * abort.
 	 */
 	if (new_target == current_target
-	   && img_type != DFU_TARGET_IMAGE_TYPE_MODEM_DELTA) {
+	   && img_type != DFU_TARGET_IMAGE_TYPE_MODEM_DELTA
+	   && current_img_num == img_num) {
```

Comment:
Should we update the above comment to mention that if the image number differs it will re-initialize?

Reply:
Sure, that's a good idea!
This API was added after the NCS v1.9.1, that's why I have not modified release changelog.
I may file a bug report in NCSK if you like...

Reply:
Issue created: NCSDK-15189

Reply:
Comment added.